### ISSUE TITLE
ELB-V2 vs ELB?

### ISSUE THREAD

**Issue by bionetmarco** [awslabs/aws-codedeploy-samples#55]
I have noticed that "_Target Group_" scripts has been added (ELB-V2). What is the difference between running CodeDeploy on a Target Group vs a regular Load Balancers? I've been scratching my head on the exact benefits to a Target Group over a Load Balancer, and whether there is value in moving to V2.

I'm trying to solve unpredictable errors in the deregistering and registering of instances from a LB during a deployment - which have been occurring recently - following 3 months of continuous deployments with no hitch. Thanks in advance.

**Comment by feverLu** [comment]
The V1 was to support the classic load balancer that routes traffic based on either application or network level information, and the V2 supports application load balancer, which is a new type that ELB supports, that routes traffic based on advanced application level information that includes the content of the request. More information can be found here: https://aws.amazon.com/elasticloadbalancing/

Basically if you want to the classic load balancer, you can continue to use the V1 script for ELB. If you have plans to use ALB, V2 script for ELB would be the sample to use.

**Comment by mvanbaak** [comment]
the ALB (ELBv2) has another advantage: you can run WAF in front of it \o/
